src/libraries/System.Text.Json/gen/Reflection/TypeExtensions.cs

```diff
@@ -16,43 +14,60 @@ public static string GetCompilableName(this Type type)
                 return GetCompilableName(type.GetElementType()) + "[]";
             }
 
-            string compilableName;
-
-            if (!type.IsGenericType)
+            if (type.IsGenericParameter)
             {
-                compilableName = type.FullName;
+                return type.Name;
             }
-            else
-            {
-                StringBuilder sb = new();
 
-                string fullName = type.FullName;
-                int backTickIndex = fullName.IndexOf('`');
+            StringBuilder sb = new();
 
-                string baseName = fullName.Substring(0, backTickIndex);
+            sb.Append("global::");
```

Comment:
I'm a bit confused by prepending this, it seems we're assuming user's class is in the `global::`?

Reply:
This is old code: https://github.com/dotnet/runtime/pull/71619/files#diff-6c3f829d936e6e251e4900b01bc9a99dfb2dd30782bfd905758f591871333db4L55
Usually the "global::" prefix is used to prevent a collision between types with similar names in similar namespaces. For example `RootNamespace.RegularNamespace.MyType` & `RegularNamespace.MyType`. "global::" is used to points to an exactly type `global::RegularNamespace.MyType`.